pkg/workloadmeta/store.go

```diff
@@ -182,6 +182,11 @@ func (s *store) Subscribe(name string, priority SubscriberPriority, filter *Filt
 			for _, cachedEntity := range entitiesOfKind {
 				entity := cachedEntity.get(sub.filter.Source())
 				if entity != nil {
+					container, ok := entity.(*Container)
```

Comment:
I don't think there should be any type-specific code in `store.go`. To me it's leaking implementation details in the generic store.